include/kodlab_mjbots_sdk/attitude.h

```diff
@@ -0,0 +1,334 @@
+              const Eigen::Vector3<Scalar> &accel_in,
+              const Eigen::Vector3<Scalar> &ang_rate_bias_in,
+              const Eigen::Quaternion<Scalar> &att_uncertainty_in,
+              const Eigen::Vector3<Scalar> &ang_bias_uncertainty_in)
+  {
+    Update(att_quat_in, ang_rate_in, accel_in);
+    ang_rate_bias_ = ang_rate_bias_in;
+    att_uncertainty_ = att_uncertainty_in;
+    ang_bias_uncertainty_ = ang_bias_uncertainty_in;
+  }
+
+  /**
+   * @brief Update this \c Attitude object's data
+   *
+   * @param pi_att
+   */
+  void Update(const ::mjbots::pi3hat::Attitude &pi_att)
+  {
+    Update({pi_att.quat.w, pi_att.quat.x, pi_att.quat.y, pi_att.quat.z},
+           {pi_att.rate_dps.x, pi_att.rate_dps.y, pi_att.rate_dps.z},
+           {pi_att.accel_mps2.x, pi_att.accel_mps2.y, pi_att.accel_mps2.z},
+           {pi_att.bias_dps.x, pi_att.bias_dps.y, pi_att.bias_dps.z},
+           {pi_att.attitude_uncertainty.w, pi_att.attitude_uncertainty.x,
+            pi_att.attitude_uncertainty.y, pi_att.attitude_uncertainty.z},
+           {pi_att.bias_uncertainty_dps.x, pi_att.bias_uncertainty_dps.y,
+            pi_att.bias_uncertainty_dps.z});
+  }
+
+  /**
+   * @brief Prints attitude, angular velocity, and linear acceleration in
+   * various representations
+   */
+  void PrintAttitude()
+  {
+    std::cout << std::fixed << std::setprecision(2)
+              << "=================================================\n"
+              << "quat_raw:\t" << get_att_quat_raw() << "\n"
+              << "    quat:\t" << get_att_quat() << "\n"
+              << "-------------------------------------------------\n"
+              << "  euler (r, p, y):\t" << get_att_euler().roll << ",\t"
+              << get_att_euler().pitch << ",\t" << get_att_euler().yaw << "\n"
+              << "ang_vel (x, y, z):\t" << get_ang_rate()(0) << ",\t"
+              << get_ang_rate()(1) << ",\t" << get_ang_rate()(2) << "\n"
+              << "lin_acc (x, y, z):\t" << get_accel()(0) << ",\t"
+              << get_accel()(1) << ",\t" << get_accel()(2) << "\n"
+              << "-------------------------------------------------\n"
+              << "rot_mat:\n" << get_att_rot_mat() << "\n"
+              << "================================================="
+              << std::endl;
+
+  }
+
+  /**
+   * @param world_offset_in world offset rotation
+   */
+  void set_world_offset(Eigen::Quaternion<Scalar> world_offset_in)
+  {
+    world_offset_ = world_offset_in;
+  }
+
+  /**
+   * @return attitude quaternion in the unmodified world frame (i.e., not yet
+   *         offset)
+   */
+  Eigen::Quaternion<Scalar> get_att_quat_raw() const { return att_quat_raw_; }
+
+  /**
+   * @return attitude quaternion in world frame
+   */
+  Eigen::Quaternion<Scalar> get_att_quat() const { return att_quat_; }
+
+  /**
+   * @return attitude rotation matrix in world frame
+   */
+  Eigen::Matrix3<Scalar> get_att_rot_mat()
+  {
```

Comment:
as mentioned above, could remove att_ but not necessary

Reply:
I'm all for leaving it.

Reply:
Class members and accessors changed to remove `att_...` in 88098f1aa2c90661cd905c056e1c3bd4b1044d2b.  Marking this as resolved.